tools/hangdetector/hang_detector.go

```diff
@@ -0,0 +1,109 @@
+package hangdetector
+
+import (
+	"io"
+	"sync/atomic"
+	"time"
+)
+
+// HangDetector ...
+type HangDetector interface {
+	Start()
+	Stop()
+	C() <-chan bool
+	WrapOutWriter(writer io.Writer) io.Writer
+	WrapErrWriter(writer io.Writer) io.Writer
+}
+
+type hangDetector struct {
+	ticker        Ticker
+	ticks         uint64
+	tickLimit     uint64
+	notificationC chan bool
+	stopC         chan bool
+
+	outWriter io.Writer
```

Comment:
I don't think we need to store this as a field anymore

Reply:
It is checked in line 60, to prevent not having an output set.

Reply:
Oh, smart!